libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/motiondetection/MotionSensor.java

```diff
@@ -39,8 +40,20 @@ public MotionSensor(Context appCtx) {
         // If no TYPE_SIGNIFICANT_MOTION is available, use alternate means to sense motion
         if (motionSensor == null) {
             motionSensor = new LegacyMotionSensor(mAppContext);
-            AppGlobals.guiLogInfo("Device has legacy motion sensor.");
         }
+        setTypeFromPrefs();
+    }
+
+    private void setTypeFromPrefs() {
```

Comment:
This is a bit awkward.  A 3 line method seems a bit useless.  Let's just inline `setTypeFromPrefs` or rename it to something like `bindSignficantMotionSensor` so that it's clear we can only get the one kind of motion sensor.